graph_snyk_cve_sync.py

```diff
@@ -217,48 +233,60 @@ def _extract_data_from_feed(self):
         for eco in self.snyk_data:
             if eco == "java" and self._parse_data_for_eco(eco):
                 logger.info("Parsing feed for Maven.")
-                self._parse_maven_data(self.snyk_data[eco])
+                self._parse_data(self.snyk_data[eco], "maven")
             elif eco == "js" and self._parse_data_for_eco(eco):
                 logger.info("Parsing feed for Npm.")
-                self._parse_npm_data(self.snyk_data[eco])
+                self._parse_data(self.snyk_data[eco], "npm")
             elif eco == "python" and self._parse_data_for_eco(eco):
                 logger.info("Parsing feed for Pypi.")
-                self._parse_pypi_data(self.snyk_data[eco])
+                self._parse_data(self.snyk_data[eco], "pypi")
             else:
                 logger.info("Ignoring the ecosystem {} from the feed".format(eco))
 
-    def _parse_maven_data(self, maven_data):
-        """Parse data for maven."""
+    def _parse_data(self, vuln_data, eco):
+        """Parse data for all eco."""
         total_vuln = 0
-        if len(maven_data) != 0:
-            self.CVE_DATA['maven'] = {}
-            for data in maven_data:
+        delta_mode = self.helper.is_delta_mode_on()
+        if len(vuln_data) != 0:
+            self.CVE_DATA[eco] = {}
+            self.DELETE_CVE_DATA[eco] = []
+            self.DELTA_FEED[eco] = []
```

Comment:
Why do we need this field? Once the records are filtered out isnt the insertion path the same?

Reply:
we dont filter the records. If we use the CVE_DATA field for this, it will contain our customized data which i wanted to avoid. wanted to keep the original snyk data. not all the additional stuff that we do it it.